enhancements/installer/assisted-installer-bare-metal-validations.md

```diff
@@ -0,0 +1,280 @@
+`bootstrapOSImage` fields, to ensure the value is a valid URL and to
+ensure the URL exists. Those checks can be split between the two
+phases, so that the URL format validation is done in
+`ValidateInstallConfig()` and the existence check is moved to
+`PlatformProvisionCheck`.
+
+The assisted installer does not need the `libvirtURI` or host BMC
+credentials, so the checks for those values could move to a new public
+function in the `pkg/types/baremetal/validation` package that can be
+invoked from
+[PlatformCredsCheck.Generate()](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformcredscheck.go#L60).
+
+Changing the BMC credential validation in the installer is only part
+of the work required to make `BareMetalHost` resources work without
+credentials. The `baremetal-operator` also [needs to be
+updated](https://github.com/metal3-io/baremetal-operator/pull/563) to
+not treat the missing credentials as an error.
+
+### Risks and Mitigations
+
+Moving some of the validation steps means that invalid data will be
+caught later in the process. Most users will run the `create cluster`
+command directly, so they will not see a significant change. When the
+installer runs under Hive, or if the user runs the phases individually
+so they can inject extra manifests, the `create cluster` step may
+happen much later in the process. Bad data will still be caught before
+the bootstrap VM is launched, though, so it should be relatively quick
+in the overall installation process.
+
+There are no security implications for this change.
+
+## Design Details
+
+### Test Plan
+
+Besides the existing and new unit tests, the existing e2e tests for
+the `baremetal` platform will continue to test the full set of
+validation rules. The test suite for the assisted installer will
+verify that no extra validation rules are applied.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
```

Comment:
Some of the sections about graduation etc don't really apply here, can we just remove them?